ops/services/app_functions/report_stream_batched_publisher/functions/common/reportingHandlers.ts

```diff
@@ -1,28 +1,56 @@
 import { Context } from "@azure/functions";
 import { DequeuedMessageItem, QueueClient } from "@azure/storage-queue";
 import fetch, { Headers, Response } from "node-fetch";
+import jwt from "jsonwebtoken";
+import crypto from "crypto";
 
 import { ENV, uploaderVersion } from "../config";
-import { ReportStreamResponse } from "./types";
+import {
+  publisherLogging,
+  ReportStreamResponse,
+  ReportStreamTokenResponse,
+} from "./types";
 import {
   deleteSuccessfullyParsedMessages,
   publishToQueue,
   reportExceptions,
 } from "./queueHandlers";
-import * as appInsights from "applicationinsights";
 
-const { FHIR_REPORT_STREAM_TOKEN, REPORT_STREAM_URL } = ENV;
+const {
+  FHIR_REPORT_STREAM_TOKEN,
+  FHIR_REPORT_STREAM_KEY,
+  REPORT_STREAM_URL,
+  REPORT_STREAM_BASE_URL,
+} = ENV;
 
-const telemetry = appInsights.defaultClient;
+export const FHIR_CLIENT_ID = "simple_report.fullelr";
 
-export async function reportToUniversalPipeline(
+export async function reportToUniversalPipelineTokenBased(
+  token: string,
+  ndjsonTestEvents: string
+): Promise<Response> {
+  const headers = new Headers({
+    authorization: `bearer ${token}`,
+    "x-api-version": uploaderVersion,
+    "content-type": "application/fhir+ndjson",
+    client: "simple_report",
+  });
+
+  return fetch(`${REPORT_STREAM_BASE_URL}/api/waters`, {
+    method: "POST",
+    headers,
+    body: ndjsonTestEvents,
+  });
+}
+
+export async function reportToUniversalPipelineSharedKey(
```

Comment:
is the plan to clean this up eventually? if so let's make a cleanup ticket

Reply:
I just cleaned up all the shared-key implementation